config/management/src/verify.rs

```diff
@@ -175,22 +186,61 @@ fn compare_genesis(
         actual_consensus_key == expected_consensus_key,
     );
 
-    let actual_validator_key = x25519_from_storage(VALIDATOR_NETWORK_KEY, storage)
+    // TODO(philiphayes): read actual shared_val_netaddr_keys from secure storage
+    let shared_val_netaddr_keys: HashMap<_, _> = iter::once((
+        TEST_SHARED_VAL_NETADDR_KEY_VERSION,
+        TEST_SHARED_VAL_NETADDR_KEY,
+    ))
+    .collect();
+
+    // decrypt and deserialize validator network address
+
+    let raw_enc_val_addrs = validator_config.validator_network_addresses;
+    let val_addrs = decrypt_addresses(
+        &validator_account,
+        &shared_val_netaddr_keys,
+        &raw_enc_val_addrs,
+    )
+    .collect::<Result<Vec<_>, _>>()
+    .map_err(|err| {
+        Error::UnexpectedError(format!("Unable to read validator network address: {}", err))
+    })?;
+    let val_pubkeys: HashSet<_> = val_addrs
+        .iter()
+        .filter_map(NetworkAddress::find_noise_proto)
+        .collect();
+
+    // deserialize full node network address
+
+    let raw_fn_addrs = validator_config.full_node_network_addresses;
+    let fn_addrs = deserialize_addresses(&validator_account, &raw_fn_addrs)
+        .collect::<Result<Vec<_>, _>>()
+        .map_err(|err| {
+            Error::UnexpectedError(format!("Unable to read fullnode network address: {}", err))
+        })?;
```

Comment:
Maybe we could pass in an error conversion to deserialize_addresses?  Seems like we need to map the error on every deserialize_addresses function call.

I'll leave that up to you though, it might make it more complicated.